### PR TITLE
General refactor and removing thor

### PR DESCRIPTION
**Pull request by manveru**

### CODE REVIEW

File: bin/bundix [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
-Bundix::CLI.start(ARGV, debug: true)
+require_relative '../lib/bundix'
+
+cli = Bundix::CLI.new(ARGV)
+command = ARGV.shift
+if cli.commands.include?(command)
+  cli.send("#{command}_options").order!
+  cli.send(command)
+elsif command # show help on unknown commands
+  puts cli.global.help
```
Comment (cstrahan) [inline bin/bundix]: Not a blocker, but might be nice to have a non-zero exit code when given an unknown command.

File: lib/bundix/manifest.rb [hunk 2]
```diff
@@ -39,7 +39,7 @@ def to_nix
       sha256 = "<%= gem.source.sha256 %>";
       fetchSubmodules = <%= gem.source.submodules %>;
       <%- elsif gem.source.type == 'gem' -%>
-      remotes = <%= gem.source.remotes.inspect %>;
+      remotes = [<%= [*gem.source.remotes].map(&:dump) * ' ' %>];
```
Comment (cstrahan) [inline lib/bundix/manifest.rb]: What's the reason for wrapping in an array and splatting here?
Also, might `Array#join` be a bit more readable that `Array#*`?

File: lib/bundix/prefetcher.rb [hunk 3]
```diff
@@ -69,7 +68,15 @@ def prefetch(source)
     case source
     when Bundix::Source::Gem
       sha = source.rubygems_org? && wrapper.gem(source.name, source.version)
-      sha ||= wrapper.url(source.urls.first) # XXX
+      return sha if sha
+
+      source.urls.find do |url|
+        begin
+          wrapper.url(url)
+        rescue Thor::Error
```
Comment (cstrahan) [inline lib/bundix/prefetcher.rb]: Now that `Wrapper#exec` isn't throwing an exception from `thor`, this `rescue` shouldn't use `Thor::Error`, right?

**Comment by manveru** [comment]
Yeah, I'll go over this again tomorrow and do some rubocop cleanup as well... it's not in a state I'm happy with yet.

**Comment by cstrahan** [comment]
Cool. Thanks a ton for cleaning this up and removing the Thor dependency. It's already looking much better!

**Comment by cstrahan** [comment]
Work is ongoing elsewhere; closing.

**Pull request closed without merging**